awx/api/views.py

```diff
@@ -2371,80 +2371,22 @@ class InventoryScriptView(RetrieveAPIView):
-                if len(obj.hosts.all()) == 0:
-                    return Response({})
-                else:
-                    all_group = data.setdefault('all', dict())
-                    smart_hosts_qs = obj.hosts.all()
-                    smart_hosts = list(smart_hosts_qs.values_list('name', flat=True))
-                    all_group['hosts'] = smart_hosts
-            else:
-                # Add hosts without a group to the all group.
-                groupless_hosts_qs = obj.hosts.filter(groups__isnull=True, **hosts_q)
-                groupless_hosts = list(groupless_hosts_qs.values_list('name', flat=True))
-                if groupless_hosts:
-                    all_group = data.setdefault('all', dict())
-                    all_group['hosts'] = groupless_hosts
-
-                # Build in-memory mapping of groups and their hosts.
-                group_hosts_kw = dict(group__inventory_id=obj.id, host__inventory_id=obj.id)
-                if 'enabled' in hosts_q:
-                    group_hosts_kw['host__enabled'] = hosts_q['enabled']
-                group_hosts_qs = Group.hosts.through.objects.filter(**group_hosts_kw)
-                group_hosts_qs = group_hosts_qs.values_list('group_id', 'host_id', 'host__name')
-                group_hosts_map = {}
-                for group_id, host_id, host_name in group_hosts_qs:
-                    group_hostnames = group_hosts_map.setdefault(group_id, [])
-                    group_hostnames.append(host_name)
-
-                # Build in-memory mapping of groups and their children.
-                group_parents_qs = Group.parents.through.objects.filter(
-                    from_group__inventory_id=obj.id,
-                    to_group__inventory_id=obj.id,
-                )
-                group_parents_qs = group_parents_qs.values_list('from_group_id', 'from_group__name', 'to_group_id')
-                group_children_map = {}
-                for from_group_id, from_group_name, to_group_id in group_parents_qs:
-                    group_children = group_children_map.setdefault(to_group_id, [])
-                    group_children.append(from_group_name)
-
-                # Now use in-memory maps to build up group info.
-                for group in obj.groups.all():
-                    group_info = dict()
-                    group_info['hosts'] = group_hosts_map.get(group.id, [])
-                    group_info['children'] = group_children_map.get(group.id, [])
-                    group_info['vars'] = group.variables_dict
-                    data[group.name] = group_info
-
-            if hostvars:
-                data.setdefault('_meta', dict())
-                data['_meta'].setdefault('hostvars', dict())
-                for host in obj.hosts.filter(**hosts_q):
-                    data['_meta']['hostvars'][host.name] = host.variables_dict
-
-        return Response(data)
+        return Response(obj.get_script_data(
+            data,
```

Comment:
hold up, what's going on with this `data` parameter?

If query param "host" is specified, this is the same as passing `--host` to an inventory script. If this option is provided, the script should return a dictionary of the host variables, and only that. It looks like you have effectively decreased the indentation of the content that used to be nested under the `else:` here, which would do some kind of crazy mixing of the full script data with an individual host's data.

...and we should test this, of course. The ability to sanely test our script-generating method using the model's method is something that I've been looking forward to.

Reply:
Yep, you're totally right.  Good eye - I missed the indentation change here.